src/item_pocket.cpp

```diff
@@ -1212,6 +1213,8 @@ void item_pocket::contents_info( std::vector<iteminfo> &info, int pocket_number,
 
     // ablative pockets have their contents displayed earlier in the UI
     if( !is_ablative() ) {
+        // Create map here for item contents, use name as key and value as number held, after traversing the entire pocket dump the map into the vector.
+        std::unordered_map<std::string, int> counted_contents;
```

Comment:
It would be better to use `std::unordered_map<item const *, int>` here, then down below use `item::stacks_with()` instead of string matching, and finally colorize only at the end.